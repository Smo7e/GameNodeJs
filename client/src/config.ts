// UdSU
// const DOMAIN = 'http://studfront';
// const PORT = 81;

// dev
const DOMAIN = "http://localhost";
const PORT = 3001;

// prod
//...

export const HOST = PORT ? `${DOMAIN}:${PORT}` : `${DOMAIN}`;

export const MEDIATOR = {
    EVENTS: {
        SERVER_ERROR: "SERVER_ERROR",
        GET_MESSAGES: "GET_MESSAGES",
        GET_SCENE: "GET_SCENE",
        GET_USER: "GET_USER",
        GET_GAMERS: "GET_GAMERS",
        GET_FRIENDS: "GET_FRIENDS",
        GET_INVITES: "GET_INVITES",
        LOGIN: "LOGIN",
        SIGNUP: "SIGNUP",
        LOGOUT: "LOGOUT",
        GET_MOBS: "GET_MOBS",
        GET_ERROR: "GET_ERROR",
        PLAY_MUSIC: "PLAY_MUSIC", 
        STOP_MUSIC: "STOP_MUSIC", 
        SET_MUSIC_VOLUME: "SET_MUSIC_VOLUME", 
        UPDATE_ARR_BULLET_TRAJECTORY: "UPDATE_ARR_BULLET_TRAJECTORY",
    },
    TRIGGERS: {},
};
