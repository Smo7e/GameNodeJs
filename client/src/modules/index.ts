import { TError } from "./Server/types";
import Server from "./Server/Server";
import Mediator from "./Mediator/Mediator";
import Store from "./Store/Store";

export type { TError };
export { Server, Mediator, Store };
