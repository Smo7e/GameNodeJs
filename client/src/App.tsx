import React, { useState } from "react";
import { HOST, MEDIATOR } from "./config";
import { Server, Mediator, Store, TError } from "./modules";
import SignUp from "./component/SignUp/SignUp";
import Game from "./modules/Game/Game";
import Menu from "./component/Menu/Menu";
import Login from "./component/Login/Login";
import Heroes from "./component/Heroes/Heroes";
import Interface from "./component/Interface/Interface";
import Lobby from "./component/Lobby/Lobby";
import ErrorMessage from "./modules/ErrorMessage/ErrorMessage";
import Parameters from "./component/Parameters/Parameters";
import Music from "./component/Parameters/Music";
import "./App.css";

export const ServerContext = React.createContext<Server>(null!);
export const MediatorContext = React.createContext<Mediator>(null!);
export const StoreContext = React.createContext<Store>(null!);

export enum EPAGES {
    SIGNUP,
    LOGIN,
    GAME,
    MENU,
    HEROES,
    LOBBY,
    PARAMETERS,
}

const MainApp = () => {
    const [epages, setEpages] = useState<EPAGES>(EPAGES.LOGIN);
    return (
        <>
            {epages === EPAGES.LOGIN ? (
                <Login epages={setEpages} />
            ) : epages === EPAGES.SIGNUP ? (
                <SignUp epages={setEpages} />
            ) : epages === EPAGES.MENU ? (
                <Menu epages={setEpages} />
            ) : epages === EPAGES.GAME ? (
                <>
                    <Game />
                    <Interface />
                </>
            ) : epages === EPAGES.HEROES ? (
                <Heroes epages={setEpages} />
            ) : epages === EPAGES.LOBBY ? (
                <Lobby epages={setEpages} />
            ) : epages === EPAGES.PARAMETERS ? (
                <Parameters epages={setEpages} />
            ) : (
                <></>
            )}
            <ErrorMessage />
            <Music />
        </>
    );
};

const App: React.FC = () => {
    const store = new Store();
    const mediator = new Mediator(MEDIATOR);
    const server = new Server(HOST, mediator, store);

    const { SERVER_ERROR } = mediator.getEventTypes();
    mediator.subscribe(SERVER_ERROR, (data: TError) => {
        console.log(data);
    });

    return (
        <StoreContext.Provider value={store}>
            <MediatorContext.Provider value={mediator}>
                <ServerContext.Provider value={server}>
                    <MainApp />
                </ServerContext.Provider>
            </MediatorContext.Provider>
        </StoreContext.Provider>
    );
};

export default App;
