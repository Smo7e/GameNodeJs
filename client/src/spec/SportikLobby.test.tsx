import { render, screen } from "@testing-library/react";
import SportikLobby from "../component/Lobby/component/SportikLobby";
import { MediatorContext, ServerContext } from "../App";
import { Mediator, Server, Store } from "../modules";
import { HOST, MEDIATOR } from "../config";

const store = new Store();
const mediator = new Mediator(MEDIATOR);
const server = new Server(HOST, mediator, store);

test("SportikLobby snapshot with gamerNumber 0", () => {
    const { asFragment } = render(
        <MediatorContext.Provider value={mediator}>
            <ServerContext.Provider value={server}>
                <SportikLobby lobby={() => {}} gamerNumber={0} />
            </ServerContext.Provider>
        </MediatorContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

test("SportikLobby snapshot with gamerNumber 1", () => {
    const { asFragment } = render(
        <MediatorContext.Provider value={mediator}>
            <ServerContext.Provider value={server}>
                <SportikLobby lobby={() => {}} gamerNumber={1} />
            </ServerContext.Provider>
        </MediatorContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

test("SportikLobby snapshot with gamerNumber 2", () => {
    const { asFragment } = render(
        <MediatorContext.Provider value={mediator}>
            <ServerContext.Provider value={server}>
                <SportikLobby lobby={() => {}} gamerNumber={2} />
            </ServerContext.Provider>
        </MediatorContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});
