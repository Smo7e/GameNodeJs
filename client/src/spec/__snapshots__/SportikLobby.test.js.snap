// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SportikLobby snapshot with gamerNumber 0 1`] = `
<DocumentFragment>
  <button
    class="arrow-2"
  />
  <button
    class="arrow-3"
  />
  <div
    class="image-Sportik"
  >
    <button
      class="button"
    >
      &lt;Спортик&gt;
    </button>
  </div>
</DocumentFragment>
`;

exports[`SportikLobby snapshot with gamerNumber 1 1`] = `
<DocumentFragment>
  <button
    class="arrow-2Friend1"
  />
  <button
    class="arrow-3Friend1"
  />
  <div
    class="image-Sportik1"
  >
    <button
      class="button"
    >
      &lt;Спортик&gt;
    </button>
  </div>
</DocumentFragment>
`;

exports[`SportikLobby snapshot with gamerNumber 2 1`] = `
<DocumentFragment>
  <button
    class="arrow-2Friend2"
  />
  <button
    class="arrow-3Friend2"
  />
  <div
    class="image-Sportik2"
  >
    <button
      class="button"
    >
      &lt;Спортик&gt;
    </button>
  </div>
</DocumentFragment>
`;
